core/modules/wsched/ScanScheduler.cc

```diff
@@ -54,11 +54,10 @@ ChunkDisk* dbgChunkDisk1 = nullptr; ///< A symbol for gdb
 ////////////////////////////////////////////////////////////////////////
 // class ScanScheduler
 ////////////////////////////////////////////////////////////////////////
-ScanScheduler::ScanScheduler()
-    : _maxRunning(32),  // FIXME: set to some multiple of system proc count.
-      _disks(),
-      _logger(LOG_GET(getName())),
-      _mutex()
+ScanScheduler::ScanScheduler(int maxThreads)
+    : _maxThreads{maxThreads},  // FIXME: set to some multiple of system proc count.
```

Comment:
The NPROC issue is more complicated than that and it may conflict with max_pid. I'd suggest deciding what the maximum can be at static initialization time and using a (ick) global variable that holds  the absolute maximum you can have. If this is the only place maxThreads makes sense then make the static a member of the class.